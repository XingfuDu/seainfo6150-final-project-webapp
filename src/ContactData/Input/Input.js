import React from 'react';
import classes from './Input.module.css';

const input = (props) => {
	let inputElement = null;
	const inputClasses = [ classes.InputElement ];
	let validationError = null;

	if (props.invalid && props.shouldValidate && props.touched) {
		inputClasses.push(classes.Invalid);
		validationError = <p className={classes.p}>Please enter a valid value!</p>;
	}

	switch (props.elementType) {
		case 'input':
			inputElement = (
				<input
					className={inputClasses.join(' ')}
					{...props.elementConfig}
					value={props.value}
					onChange={props.changed}
				/>
			);
			break;
		case 'textarea':
			inputElement = (
				<textarea
					className={inputClasses.join(' ')}
					{...props.elementConfig}
					value={props.value}
					onChange={props.changed}
				/>
			);
			break;
		case 'select':
			inputElement = (
				<select className={inputClasses.join(' ')} value={props.value} onChange={props.changed}>
					{props.elementConfig.options.map((option, index) => (
						<option key={option.value} value={option.value}>
							{option.displayValue}
						</option>
					))}
				</select>
			);
			break;
		case 'checkbox':
			inputElement = (
				<input
					className={classes.CheckBox}
					{...props.elementConfig}
					value={props.value}
					onChange={props.changed}
					id={props.label}
				/>
			);
			break;
		case 'radio':
			inputElement = (
				<div className={classes.Radio}>
					<label htmlFor={props.labelText[0]} className={classes.Radio} >
						{props.labelText[0]}
					</label>
					<input
						className={classes.Radio}
						{...props.elementConfig}
						value={props.labelText[0]}
						onChange={props.changed}
						id={props.labelText[0]}
						name={props.name}
					/>
					<label htmlFor={props.labelText[1]} className={classes.Radio}>
						{props.labelText[1]}
					</label>
					<input
						className={classes.Radio}
						{...props.elementConfig}
						value={props.labelText[1]}
						onChange={props.changed}
						id={props.labelText[1]}
						name={props.name}	
					/>
					<label htmlFor={props.labelText[2]} className={classes.Radio}>
						{props.labelText[2]}
					</label>
					<input
						className={classes.Radio}
						{...props.elementConfig}
						value={props.labelText[2]}
						onChange={props.changed}
						id={props.labelText[2]}
						name={props.name}
					/>
				</div>
			);
			break;
		default:
			inputElement = (
				<input
					className={inputClasses.join(' ')}
					{...props.elementConfig}
					value={props.value}
					onChange={props.changed}
				/>
			);
	}
	return (
		<div className={classes.Input}>
			<label htmlFor={props.label} className={classes.Label}>
				{props.labelContent}
			</label>
			{inputElement}
			{validationError}
		</div>
	);
};

export default input;
