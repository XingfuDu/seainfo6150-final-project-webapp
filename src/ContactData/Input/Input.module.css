.Input {
    width: 100%;
    padding: 10px;
    box-sizing: border-box;
}

.Label {
    font-weight: bold;
    margin-bottom: 8px;
}

.InputElement {
    display: block;
    outline: none;
    border: 1px solid #ccc;
    background-color: white;
    font: inherit;
    padding: 6px 10px;
    width: 100%;
    box-sizing: border-box;
}

.CheckBox {
    border: 1px solid #ccc;
    margin-left: 20px;
    width: 20px;
    height: 20px;
    display: inline;
    box-sizing: border-box;
}

.InputElement:focus {
    outline: none;
    background-color: #ccc;
}

.Invalid {
    border: 1px solid #ee6c4d;
    background-color: #FDA49A;
}

.p {
	color: #3d5a80;
}