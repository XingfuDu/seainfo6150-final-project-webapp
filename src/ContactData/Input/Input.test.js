import React from 'react';
import Input from './Input';
import { render } from "@testing-library/react";

describe('Input tests', () => {
	it('renders correctly', () => {
		const { container } = render(<Input label="test" labelContent="test"/>);
		expect(container).toMatchSnapshot();
	});
});
