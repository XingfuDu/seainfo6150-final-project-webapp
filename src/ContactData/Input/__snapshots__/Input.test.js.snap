// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input tests renders correctly 1`] = `
<div>
  <div
    class="Input"
  >
    <label
      class="Label"
      for="test"
    >
      test
    </label>
    <input
      class="InputElement"
      value=""
    />
  </div>
</div>
`;
